app/routes/events/components/EventList.js

```diff
@@ -1,157 +1,168 @@
-  style: any
 };
 
-function Attendance({
-  registrationCount,
-  totalCapacity,
-  style
-}: AttendanceProps) {
-  // @todo choose pill color based on capacity
-  return (
-    <Pill style={{ ...style, whiteSpace: 'nowrap' }}>
-      {`${registrationCount} / ${totalCapacity}`}
-    </Pill>
-  );
-}
-type EventItemProps = {
-  event: any,
-  showTags?: boolean
-};
-export function EventItem({ event, showTags = true }: EventItemProps) {
-  return (
-    <div
-      style={{ borderColor: colorForEvent(event.eventType) }}
-      className={styles.eventItem}
-    >
-      <div>
-        <Link to={`/events/${event.id}`}>
-          <h3 className={styles.eventItemTitle}>{event.title}</h3>
-          {event.totalCapacity > 0 && (
-            <Attendance
-              registrationCount={event.registrationCount}
-              totalCapacity={event.totalCapacity}
-              style={{ marginLeft: '5px', color: 'black' }}
-            />
-          )}
-        </Link>
-
-        <div className={styles.eventTime}>
-          <Time time={event.startTime} format="ll HH:mm" />
-          {` • ${event.location}`}
-        </div>
-        {showTags && (
-          <Flex wrap className={styles.tagList}>
-            {event.tags.map((tag, index) => (
-              <Tag key={index} tag={tag} small />
-            ))}
-          </Flex>
-        )}
-      </div>
-
-      <div className={styles.companyLogo}>
-        {event.cover && <Image src={event.cover} />}
-      </div>
-    </div>
-  );
-}
-
-type EventListGroupProps = {
+const EventListGroup = ({
+  name,
+  field = 'startTime',
+  events = []
+}: {
   name: string,
-  events: Array</*TODO: Event */ any>
-};
-
-function EventListGroup({ name, events = [] }: EventListGroupProps) {
+  field: string,
+  events: Array<Event>
+}) => {
   return isEmpty(events) ? null : (
     <div className={styles.eventGroup}>
       <h2 className={styles.heading}>{name}</h2>
       {events.map((event, i) => (
-        <EventItem key={i} event={event} />
+        <EventItem key={i} event={event} field={field} />
       ))}
     </div>
   );
-}
+};
 
 type EventListProps = {
-  events: Array<any>,
-  actionGrant: /* TODO: ActionGrant */ any,
-  icalToken: /* TODO: IcalToken */ string,
+  events: Array<Event>,
+  actionGrant: ActionGrant,
+  icalToken: string,
   showFetchMore: boolean,
   fetchMore: () => Promise<*>
 };
 
-// $FlowFixMe
-const EventList = (props: EventListProps) => {
-  const events = groupEvents(props.events);
-  const { icalToken, showFetchMore, fetchMore } = props;
-  return (
-    <div className={styles.root}>
-      <Helmet title="Arrangementer" />
-      <Toolbar actionGrant={props.actionGrant} />
-      <EventListGroup name="Denne uken" events={events.currentWeek} />
+type Option = {
+  value: Event => Event,
+  label: string,
+  field: string
+};
 
-      <EventListGroup name="Neste uke" events={events.nextWeek} />
+type State = {
+  selectedOption: Option
+};
+
+// $FlowFixMe
```

Comment:
Yeah, it results in a large ``mapStateToProps   connect()` error. I've seen a lot of people using `// $FlowFixMe` for this exact purpose, so it would be good to find the solution here.

Reply:
We did a quick search, we ignore Flow typings 42 times. Some of them shouldn't be ignored, it just looks like laziness/not knowing how to use flow typings. But in cases like this I agree.